app/src/views/features/sessions/types.ts

```diff
@@ -0,0 +1,23 @@
+import { SourceKey, SourceOperator } from "types";
+
+export enum AutoRecordingMode {
+  CUSTOM = "custom",
+  ALL_PAGES = "allPages",
+}
+
+export type PageSource = {
+  id?: string;
+  key: SourceKey;
+  value: string;
+  isActive?: boolean;
+  operator: SourceOperator;
```

Comment:
`id` and `isActive` should be mandatory now?

Reply:
`id` will be optional, since it's assigned when we are creating the page source (while saving)